### PR TITLE
Ignore socket hangup when ending connection also with ssl (#1344).

### PR DESCRIPTION
**Pull request by pasieronen**
See issue #1344 for details.

### CODE REVIEW

File: test/unit/connection/error-tests.js [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+})
+
+
+suite.test('connection does not emit ECONNRESET errors during disconnect also when using SSL', function (done) {
+  // our fake postgres server, which just responds with 'S' to start SSL
+  var socket
+  var server = net.createServer(function (c) {
+    socket = c
+    c.once('data', function (data) {
+      c.write(new Buffer('S', 'utf8'))
```
Comment (charmander) [inline test/unit/connection/error-tests.js]: You can use `Buffer.from`; the versions of Node pg 7 supports all support it.

File: lib/connection.js [hunk 2]
```diff
@@ -97,11 +97,17 @@ Connection.prototype.connect = function (port, host) {
       NPNProtocols: self.ssl.NPNProtocols
     })
     self.attachListeners(self.stream)
-    self.emit('sslconnect')
 
     self.stream.on('error', function (error) {
```
Comment (charmander) [inline lib/connection.js]: Can this reuse the function from earlier instead of duplicating it?

**Comment by ubreddy** [comment]
any idea when this will be merged and new version release?

**Comment by brianc** [comment]
This is fantastic. Sorry for the delay on merging this. I'll push a new minor version out right now w/ the fix.

**Comment by brianc** [comment]
Published `pg@7.4.1`

**Pull request merged**